### ISSUE TITLE
rotate_extrude doesn't handle 2D polygons touching the Y axis in one point

### ISSUE THREAD

**Issue by sbguthery** [openscad/openscad#1081]
Here's the OpenSCAD code:

dxffile="207a.dxf";
difference(){
translate([0,0,-10])rotate_extrude(convexity=10) import(dxffile);
translate([140,0,10])cube([300,600,300],center=true);
};

The DXF file is at www.scdk.com/207a.dxf

Here's the Preview:

![rotatepic](https://cloud.githubusercontent.com/assets/10210456/5477300/2a3866fa-85f8-11e4-8659-3361f7edcbc4.jpg)

Here's the assertion violation during Render:

Compiling design (CSG Tree generation)...
Rendering Polygon Mesh using CGAL...
CGAL error in CGAL_Nef_polyhedron3(): CGAL ERROR: assertion violation!
Expr: e->incident_sface() != SFace_const_handle()
File: /data/OpenSCAD/libraries-mingw64-master/mxe-w64/usr/x86_64-w64-mingw32.static/include/CGAL/Nef_S2/SM_const_decorator.h
Line: 326
Geometries in cache: 5
Geometry cache size in bytes: 7177856
CGAL Polyhedrons in cache: 2
CGAL cache size in bytes: 0
Total rendering time: 0 hours, 0 minutes, 2 seconds
Rendering finished.
##

**Comment by stuartpb** [comment]
I'm getting a similar error to this, but with an imported STL that's not being extruded.

Observed with version 2015.03-2

Steps to reproduce:

1. Download this STL model: https://www.thingiverse.com/download:3144674
2. Attempt to render this:

```scad
translate([-11.2501,-20.001,0.001]) rotate (155) import("viveBUCKLE.stl");
linear_extrude(height=5) square(size=[50,30], center=true);
```

Error output:

```
Compiling design (CSG Tree generation)...
Rendering Polygon Mesh using CGAL...
ERROR: CGAL error in CGAL_Nef_polyhedron3(): CGAL ERROR: assertion violation! Expr: e->incident_sface() != SFace_const_handle() File: /opt/mxe/usr/x86_64-w64-mingw32.static/include/CGAL/Nef_S2/SM_const_decorator.h Line: 326 
Geometries in cache: 90
Geometry cache size in bytes: 3879560
CGAL Polyhedrons in cache: 44
CGAL cache size in bytes: 26726448
Total rendering time: 0 hours, 0 minutes, 0 seconds
   Top level object is a 3D object:
   Simple:        yes
   Vertices:        8
   Halfedges:      24
   Edges:          12
   Halffacets:     12
   Facets:          6
   Volumes:         2
Rendering finished.
```

The final render contains only the extruded shape and not the imported model.

**Comment by MichaelAtOz** [comment]
@stuartpb Yes, the STL was non-manifold. Also the code above is NOT rotate_extrude(), so it would not be an instance of this issue.

**Comment by kintel** [comment]
fwiw, this works when enabling Manifold

**Comment by kintel** [comment]
https://github.com/openscad/openscad/blob/master/tests/data/scad/3D/features/rotate_extrude-touch-vertex.scad
